v4/core/base_service.go

```diff
@@ -451,3 +493,157 @@ func getErrorMessage(responseMap map[string]interface{}, statusCode int) string
 	// for the status code.
 	return http.StatusText(statusCode)
 }
+
+// EnableRetries will construct a "retryable" HTTP Client with the specified
+// configuration, and then set it on the service instance.
+// If maxRetries and/or maxRetryInterval are specified as 0, then default values
+// are used instead.
+func (service *BaseService) EnableRetries(maxRetries int, maxRetryInterval time.Duration) {
+	client := NewRetryableHTTPClient()
+	if maxRetries > 0 {
+		client.RetryMax = maxRetries
+	}
+	if maxRetryInterval > 0 {
+		client.RetryWaitMax = maxRetryInterval
+	}
+
+	service.SetHTTPClient(client.StandardClient())
+}
+
+// DisableRetries will disable automatic retries by constructing a new
+// default (non-retryable) HTTP Client instance and setting it on the service.
+func (service *BaseService) DisableRetries() {
+	service.SetHTTPClient(DefaultHTTPClient())
+}
+
+// DefaultHTTPClient returns a non-retryable http client with default configuration.
+func DefaultHTTPClient() *http.Client {
+	return cleanhttp.DefaultPooledClient()
+}
+
+// httpLogger is a shim layer used to allow the Go core's logger to be used with the retryablehttp interfaces.
+type httpLogger struct {
+}
+
+func (l *httpLogger) Printf(format string, inserts ...interface{}) {
+	GetLogger().Log(LevelDebug, format, inserts...)
+}
+
+// NewRetryableHTTPClient returns a new instance of go-retryablehttp.Client
+// with a default configuration that supports Go SDK usage.
+func NewRetryableHTTPClient() *retryablehttp.Client {
```

Comment:
I initially made this private, thinking that EnableRetries() would be the only way that an SDK user would be able to enable retries.   But then I decided to make this public to give the SDK user more freedom in configuring a retryable client for rare cases where EnableRetries() isn't enough to satisfy their requirements.  For example, if the SDK user wanted to use their own "CheckRetry" and/or "Backoff" callback functions, they can use NewRetryableHTTPClient(), then set the various fields however they want.   With the way in which the retryable client is hung off the BaseService.Client field (i.e. it's hidden inside the BaseService.Client's Transport field), it would be difficult for an SDK user to get ahold of that after calling EnableRetries().